lib/awspec/helper/finder/nlb.rb

```diff
@@ -0,0 +1,57 @@
+module Awspec::Helper
+  module Finder
+    module Nlb
+      def find_nlb(id)
+        res = elbv2_client.describe_load_balancers({ names: [id] })
+        res.load_balancers.select do |lb|
+          lb.type == 'network'
+        end.single_resource(id)
+      rescue
+        return nil
+      end
+
+      def select_nlb_by_vpc_id(vpc_id)
+        res = elbv2_client.describe_load_balancers
+        res.load_balancers.select do |lb|
+          lb.vpc_id == vpc_id && lb.type == 'network'
+        end
+      end
+
+      def find_nlb_listener(arn)
+        res = elbv2_client.describe_listeners({ listener_arns: [arn] })
+        res.listeners.single_resource(arn)
+      rescue
+        return nil
+      end
+
+      def find_nlb_target_group(id)
+        res = elbv2_client.describe_target_groups({ names: [id] })
```

Comment:
I expect the original motivation to filter target groups on protocol (https://github.com/k1LoW/awspec/commit/ab499b44d4b171da47e7cd3c105d46d16ac722d4#diff-0822f8d08c87e802662eecb88a260475R30) was to be able to reuse the same target group name for more than one target group: Groups that differed only by protocol and port - in which case the default t.g. that an otherwise unqualified name intended to refer to was the HTTP or HTTPS one.

But in the case of target groups that are named distinctly, restricting the result to *only* HTTP or HTTPS makes it difficult to test a target group that uses protocol "TCP" (as an example). The NLB code here preserves the preference for HTTP/S - but rescues itself if it finds no groups with those protocols - and instead returns the unfiltered single_resource() result, if any.